open_alchemy/cli.py

```diff
@@ -0,0 +1,82 @@
+"""Define the CLI module."""
+import argparse
+import logging
+import pathlib
+
+from open_alchemy import PackageFormat
+from open_alchemy import build_json
+from open_alchemy import build_yaml
+
+# Configure the logger.
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+def main():
+    """Define the CLI entrypoint."""
+    args = build_application_parser()
+    try:
+        args.func(args)
+    except Exception as exc:  # pylint: disable=broad-except
+        logging.error("Cannot perform the operation: %s.", exc)
+
+
+def build_application_parser():
+    """Build the main parser and subparsers."""
+    # Define the top level parser.
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(title="subcommands", help="subcommand help")
+
+    # Define the parser for the "build" subcommand.
+    build_parser = subparsers.add_parser(
+        "build",
+        description="Build a package with the SQLAlchemy models.",
+        help="build a package",
+    )
+    build_parser.add_argument(
+        "specfile", type=str, help="specify the specification file"
+    )
+    build_parser.add_argument("name", type=str, help="specify the package name")
+    build_parser.add_argument("output", type=str, help="specify the output directory")
+    build_parser.add_argument(
+        "-f",
+        "--format",
+        type=str.lower,
+        choices=["sdist", "wheel"],
+        help="limit the format to either sdist or wheel",
+    )
+    build_parser.set_defaults(func=build)
+
+    # Return the parsed arguments for a particular command.
+    return parser.parse_args()
+
+
+def build(args):
+    """Define the build subcommand."""
+    # Check the specfile.
+    specfile = pathlib.Path(args.specfile)
+    if not specfile.exists():
+        raise ValueError(f"the specfile '{specfile}' cannot be found")
+
+    # Select the builder method.
+    builders = {
+        ".json": build_json,
+        ".yml": build_yaml,
+        ".yaml": build_yaml,
+    }
+    ext = specfile.suffix.lower()
+    if ext not in builders.keys():
+        raise ValueError(
+            f"specification format not supported: '{args.specfile}' "
+            "(should be json or yaml)"
```

Comment:
It would be excellent to describe what was received versus what was expected. I.e. `"received extension <extension>, supported extensions: .yaml, .yml, .json (case insensitive)"`

Reply:
Addressed ✅